include/libdnf/conf/option_binds.hpp

```diff
@@ -1,83 +1,91 @@
+        const char * get_name() const noexcept override { return "Exception"; }
+        const char * get_description() const noexcept override { return "OptionBinds exception"; }
     };
-    struct OutOfRange : public Exception {
-        OutOfRange(const std::string & id) : Exception(id) {}
-        const char * what() const noexcept override;
+
+    class OptionNotFound : public Exception {
+    public:
+        using Exception::Exception;
+        const char * get_name() const noexcept override { return "OptionNotFound"; }
+        const char * get_description() const noexcept override { return "Option not found"; }
     };
-    struct AlreadyExists : public Exception {
-        AlreadyExists(const std::string & id) : Exception(id) {}
-        const char * what() const noexcept override;
+
+    class OptionAlreadyExists : public Exception {
+    public:
+        using Exception::Exception;
+        const char * get_name() const noexcept override { return "OptionAlreadyExists"; }
+        const char * get_description() const noexcept override { return "Option already exists"; }
     };
 
+    /// Extends the option with user-defined functions for processing a new value and converting value to a string.
+    /// It is used as additional level of processing when the option is accesed by its text name.
     class Item final {
     public:
-        typedef std::function<void(Option::Priority, const std::string &)> NewStringFunc;
-        typedef std::function<const std::string & ()> GetValueStringFunc;
+        using NewStringFunc = std::function<void(Option::Priority, const std::string &)>;
+        using GetValueStringFunc = std::function<const std::string &()>;
 
-        Option::Priority getPriority() const;
-        void newString(Option::Priority priority, const std::string & value);
-        std::string getValueString() const;
-        bool getAddValue() const;
+        Option::Priority get_priority() const;
+        void new_string(Option::Priority priority, const std::string & value);
+        std::string get_value_string() const;
+        bool get_add_value() const;
 
     private:
         friend class OptionBinds;
 
-        Item(Option & option, const NewStringFunc & newString,
-             const GetValueStringFunc & getValueString, bool addValue);
-        Item(Option & option, NewStringFunc && newString,
-             GetValueStringFunc && getValueString, bool addValue);
-        Item(Option & option);
+        Item(Option & option, NewStringFunc new_string, GetValueStringFunc get_value_string, bool add_value);
+        explicit Item(Option & option);
         Option * option;
-        NewStringFunc newStr;
-        GetValueStringFunc getValueStr;
-        bool addValue{false}; // hint that new value be added
+        NewStringFunc new_str;
+        GetValueStringFunc get_value_str;
+        bool add_value{false};  // hint that new value be added/appended
```

Comment:
Not sure that i understand it correctly. What about:
`bool added_value{false};  // hint that a new value was added or appended`

Reply:
@j-mracek The value is set in constructor and is not used in the libdnf. It is only information for application. The application can read the value using getter `get_add_value`. True means it is an "append" option -> new value is not set but added/appended to the existing one. 
Maybe the name of the variable and name of the getter is confusing.

I can rename it. What do you think about this?
add_value -> is_append_option
get_add_value() -> get_is_append_option()

Reply:
The rename is a good solution.